support/Pipelines/Bbh/Inspiral.py

```diff
@@ -401,6 +421,50 @@ def start_inspiral(
         " 'eccentricity_control'. Choose which of the two to perform next."
     )
 
+    assert (lev is not None) or (
+        (refinement_level is not None) and (polynomial_order is not None)
+    ), (
+        "Resolution not fully specified. This could be either missing 'lev' or"
+        " missing one of either 'refinement_level' or 'polynomial_order'. The"
+        " suggested lev is lev = "
+        + str(DEFAULT_LEV["Lev number"])
+        + " (L = "
+        + str(DEFAULT_LEV["refinement level"])
+        + ", P = "
+        + str(DEFAULT_LEV["polynomial order"])
+        + ")."
+    )
+
+    if (refinement_level is not None) or (polynomial_order is not None):
+        assert (
+            lev is None
+            and (refinement_level is not None)
+            and (polynomial_order is not None)
+        ), (
+            "Specify a resolution to run. Either 'lev' or both"
+            " 'refinement_level' and 'polynomial_order'."
+        )
+
+    selected_lev = None
+    if lev is not None:
+        assert (refinement_level is None) and (polynomial_order is None), (
+            "Cannot specify both 'lev' together with 'refinement_level' and"
+            " 'polynomial_order'."
+        )
+        for element in INSPIRAL_LEVS:
+            if lev == element["Lev number"]:
+                selected_lev = element
+        assert (
+            selected_lev is not None
+        ), "Could not select lev from 'INSPIRAL_LEVS'. Possibily undefined."
+        refinement_level = selected_lev["refinement level"]
+        polynomial_order = selected_lev["polynomial order"]
+
+    assert (refinement_level is not None) and (polynomial_order is not None), (
+        "Either 'refinement_level' or 'polynomial_order' is not properly"
+        " specified."
+    )
```

Comment:
I think most of this added code and the asserts can be trimmed down to this:

```py
if lev is not None:
    assert refinement_level is None and polynomial_order is None, "..."
    selected_lev = INSPIRAL_LEVS[lev]
    # set L, P
else:
    assert refinement_level is not None and polynomial_order is not None, "..."
```

I find this if-else with nested asserts easier to understand.